packages/camera/camera_android_camerax/test/observer_test.dart

```diff
@@ -0,0 +1,114 @@
+      TestInstanceManagerHostApi.setup(null);
+    });
+
+    test('HostApi create makes call to create Observer instance', () {
+      final MockTestObserverHostApi mockApi = MockTestObserverHostApi();
+      TestObserverHostApi.setup(mockApi);
+      TestInstanceManagerHostApi.setup(MockTestInstanceManagerHostApi());
+
+      final InstanceManager instanceManager = InstanceManager(
+        onWeakReferenceRemoved: (_) {},
+      );
+
+      final Observer<dynamic> instance = Observer<dynamic>(
+        instanceManager: instanceManager,
+        onChanged: (Object value) {},
+      );
+
+      verify(mockApi.create(
+        instanceManager.getIdentifier(instance),
+      ));
+    });
+
+    test(
+        'HostAPI create makes Observer instance that throws assertion error if onChanged receives unexpected parameter type',
+        () {
+      final MockTestObserverHostApi mockApi = MockTestObserverHostApi();
+      TestObserverHostApi.setup(mockApi);
+      TestInstanceManagerHostApi.setup(MockTestInstanceManagerHostApi());
+
+      final Observer<String> cameraStateObserver =
+          Observer<String>.detached(onChanged: (Object value) {});
+
+      expect(
+          () => cameraStateObserver.onChanged(
+              CameraState.detached(type: CameraStateType.pendingOpen)),
+          throwsAssertionError);
+    });
+
+    test(
+        'FlutterAPI onChanged makes call with expected parameter to Observer instance onChanged callback',
+        () {
+      final InstanceManager instanceManager = InstanceManager(
+        onWeakReferenceRemoved: (_) {},
+      );
+
+      const int instanceIdentifier = 0;
+      late final Object? callbackParameter;
+      final Observer<CameraState> instance = Observer<CameraState>.detached(
+        onChanged: (Object value) {
+          callbackParameter = value;
+        },
+        instanceManager: instanceManager,
+      );
+      instanceManager.addHostCreatedInstance(
+        instance,
+        instanceIdentifier,
+        onCopy: (Observer<CameraState> original) =>
+            Observer<CameraState>.detached(
+          onChanged: original.onChanged,
+          instanceManager: instanceManager,
+        ),
+      );
+
+      final ObserverFlutterApiImpl flutterApi = ObserverFlutterApiImpl(
+        instanceManager: instanceManager,
+      );
+
+      const CameraStateType cameraStateType = CameraStateType.closed;
+
+      final CameraState value = CameraState.detached(
+        instanceManager: instanceManager,
+        type: cameraStateType,
+      );
+      const int valueIdentifier = 11;
+      instanceManager.addHostCreatedInstance(
+        value,
+        valueIdentifier,
+        onCopy: (_) => CameraState.detached(
+          instanceManager: instanceManager,
+          type: cameraStateType,
+        ),
+      );
+
+      flutterApi.onChanged(
+        instanceIdentifier,
+        valueIdentifier,
+      );
+
+      expect(callbackParameter, value);
+    });
+  });
```

Comment:
Can you add a test to make sure createFromInstances copies the values correctly? 

Same for other classes if a similar method exists.

Reply:
This is handled in the host API create tests. createFromInstances calls through to the mockApis so when we test that call, we ensure createFromInstances is handling the values correctly